desc/objectives/_geometry.py

```diff
@@ -357,3 +363,147 @@ def compute(self, *args, **kwargs):
+    """Penalize the minimum distance between the plasma and a surounding surface.
+
+    Computes the minimum distance from each point on the plasma grid to a point on the
+    surface grid. For dense grids, this will approximate the global min, but in general
+    will only be an upper bound on the minimum separation between the plasma and the
+    surrounding surface.
+
+    Parameters
+    ----------
+    surface : Surface
+        Bounding surface to penalize distance to.
+    eq : Equilibrium, optional
+        Equilibrium that will be optimized to satisfy the Objective.
+    target : float, ndarray, optional
+        Target value(s) of the objective.
+        len(target) must be equal to Objective.dim_f
+    weight : float, ndarray, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        len(weight) must be equal to Objective.dim_f
+    normalize : bool
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool
+        Whether target should be normalized before comparing to computed values.
+        if `normalize` is `True` and the target is in physical units, this should also
+        be set to True.
+    surface_grid : Grid, ndarray, optional
+        Collocation grid containing the nodes to evaluate surface geometry at.
+    plasma_grid : Grid, ndarray, optional
+        Collocation grid containing the nodes to evaluate plasma geometry at.
+    name : str
+        Name of the objective function.
+    """
+
+    _scalar = False
+    _linear = False
+    _units = "(m)"
+    _print_value_fmt = "Minimum plasma-surface distance: {:10.3e} "
+
+    def __init__(
+        self,
+        surface,
+        eq=None,
+        target=0,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        surface_grid=None,
+        plasma_grid=None,
+        name="vessel distance",
+    ):
+        self._surface = surface
+        self._surface_grid = surface_grid
+        self._plasma_grid = plasma_grid
+        super().__init__(
+            eq=eq,
+            target=target,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, eq, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        if self._surface_grid is None:
+            self._surface_grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+        if self._plasma_grid is None:
+            self._plasma_grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+        if not np.allclose(self._surface_grid.nodes[:, 0], 1):
```

Comment:
why does the surface_grid even check rho points? should they not just be theta,zeta points as the surface is a FourierRZToroidalSurface, and so does not really have any rho dependence?

Reply:
In theory yes, any rho coordinate will be ignored. So this check is more to make the user aware that there might be duplicate points which are wasteful (ie, same theta/zeta but different rho are really the same point).